arenarobot/service/processor/apriltag_detector.py

```diff
@@ -0,0 +1,178 @@
+"""
+apriltag_detector.py: reads frames, detects apriltags, and returns pose.
+
+Created by John Pedraza on 6/7/22.
+
+Copyright (c) 2022, The CONIX Research Center
+All rights reserved.
+
+This source code is licensed under the BSD-3-Clause license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+from json import JSONEncoder, dumps, loads
+import numpy as np
+import cv2
+from paho.mqtt.client import MQTTMessage
+from transformations import euler_matrix, quaternion_matrix
+from dt_apriltags import Detector
+from arenarobot.service.processor import ArenaRobotServiceProcessor
+from scipy.spatial.transform import Rotation as R
+
+# pylint: disable=too-many-instance-attributes
+class ArenaRobotServiceProcessorApriltagDetector(ArenaRobotServiceProcessor):
+    """Apriltag Detector processor class for the ARENA."""
+
+    DEVICE_INSTANCE_PROCESSOR_TYPE = "apriltag_detector"
+
+    # pylint: disable=too-many-arguments
+    # pylint: disable=too-many-statements
+    def __init__(self, processor_apriltag_detector_topic: str,
+                 processor_apriltag_detector_instance_name: str = None,
+                 scale_factor: int = 1,
+                 camera_resolution = None,
+                 camera_params = None,
+                 dist_params = None,
+                 apriltag_family = None,
+                 apriltag_locations = None,
```

Comment:
If parameters are required, then they should be listed before parameters with default values and not have a default.

Reply:
Having a video capture is required, but how are we supposed to pass those via the json file and not use a default value? I suppose we could pass the /dev/video# and then use that in cv2.VideoCapture(#)? But this format doesn't always work (see: all the struggle we had with non-USB cameras and having to use the picamera2 library). 

(Also, the video capture param isn't in the selected code for this PR conversation).

Reply:
IIRC OpenCV [let's you pass in a full path string](https://docs.opencv.org/3.4/d8/dfe/classcv_1_1VideoCapture.html) (`/dev/video0`) instead of just `0`. Let's assume it will be a string like that for now. I think the MJPEG stream will also work that way.